### PR TITLE
Add auto-save transcripts to disk

### LINKED ISSUE #48: Auto-save transcripts to a user-chosen folder
Image

## Summary

Automatically write transcripts to a user-chosen folder after transcription completes. No manual export step needed.

**Origin:** Community feedback (Slack, Apr 3 2026)
> "Once something is transcribed, allow me to save the transcription automatically to a folder of my choosing. So like I set the 'Save all transcripts to disk' path in the app's settings, and then after anything is transcribed, it writes like `<custom-path>/<date>-<time>-transcription.md`"

### Settings UI
- **Toggle:** Auto-save transcripts to disk (off by default)
- **Folder picker:** Choose destination folder (persisted via security-scoped bookmark)
- **Format:** Markdown (default), TXT, SRT, VTT, JSON, DOCX, PDF
- **Scope:** All / File transcription only / Dictation only

### Filename convention
- File transcription: `YYYY-MM-DD-HHmmss-<source-filename>.md`
- YouTube: `YYYY-MM-DD-HHmmss-<video-title>.md`
- Dictation: `YYYY-MM-DD-HHmmss-dictation.md`

### Behavior
- Fires after transcription completes (both dictation and file transcription)
- Uses existing `ExportService` for format rendering
- Security-scoped bookmark for persistent folder access across launches
- Timestamp naming avoids overwrite conflicts

### Open questions
- Should dictation auto-save have a minimum word count threshold to avoid tiny files?
- Should the filename template be user-configurable, or is a sensible default enough for v1?

## Implementation notes
- Leverages existing `ExportService` (all formats already supported)
- New settings: `autoSaveEnabled`, `autoSaveFolderBookmark`, `autoSaveFormat`, `autoSaveScope`
- Hook into transcription completion path in `TranscriptionService` and `DictationService`

### PR DESCRIPTION
**Pull request by moona3k**
## Summary
- Adds a new **auto-save** setting that automatically writes completed transcriptions to a user-chosen folder
- Supports 5 text formats: Markdown (default), Plain Text, SRT, VTT, JSON
- Timestamped filenames (`2026-04-03-134900-interview.md`) with deduplication
- Settings UI with toggle, format picker, and folder chooser (NSOpenPanel)
- 18 new tests covering all formats, edge cases, and bookmark lifecycle
- 1113 total tests passing (up from 1092)

## How it works
1. User enables "Auto-save transcripts to disk" in Settings → Transcription
2. Chooses a folder and format
3. After any file/YouTube transcription completes, a file is written automatically
4. Failures are logged silently — never interrupts the user

Screenshot 2026-04-03 at 6 23 13 PM

Screenshot 2026-04-03 at 6 24 20 PM

## Design decisions
- **Text formats only** — PDF/DOCX excluded (binary, `@MainActor`-heavy, not useful for Obsidian/file-based workflows)
- **Transcriptions only** — Dictation auto-save deferred (short snippets would create noise)
- **URL bookmarks** for persistent folder access across app relaunches
- **No injection** — follows existing `ExportService()` inline creation pattern

## Test plan
- [x] All 5 formats write correct content to disk
- [x] Disabled state does nothing
- [x] No folder configured does nothing
- [x] Filename contains date + sanitized source name
- [x] Duplicate filenames get `(1)`, `(2)` suffix
- [x] Deleted folder doesn't crash
- [x] Invalid bookmark data returns nil
- [x] Invalid format string falls back to Markdown
- [x] Bookmark storage and resolution round-trips
- [x] Full suite: 1113 tests, 0 failures

Closes #48

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Release Notes

* **New Features**
  * Added auto-save functionality to automatically export completed transcriptions to disk
  * Introduced transcript format selection (TXT, Markdown, SRT, VTT, JSON)
  * Added settings to enable/disable auto-save and configure output folder location
  * Transcripts are now automatically saved when auto-save is enabled

### CODE REVIEW

File: Tests/MacParakeetTests/Services/AutoSaveServiceTests.swift [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+    func testBuildFileURLSanitizesFilename() {
+        configureAutoSave(enabled: true, format: .md)
+        let transcription = makeTranscription(fileName: "my/weird:file.mp3")
+        let service = makeService()
+
+        let url = service.buildFileURL(for: transcription, format: .md, in: tempDir)
+        XCTAssertFalse(url.lastPathComponent.contains("/"))
+        XCTAssertFalse(url.lastPathComponent.contains(":"))
+        XCTAssertTrue(url.lastPathComponent.hasSuffix(".md"))
+    }
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Tests/MacParakeetTests/Services/AutoSaveServiceTests.swift]: **Accepted.** Good catch — `lastPathComponent` strips `/` by definition so the original assertion was tautological. Fixed in a809c7d: now asserts the parent directory matches `tempDir` to actually verify path separators were sanitized.

File: Sources/MacParakeetCore/Services/AutoSaveService.swift [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+                defaults.set(refreshed, forKey: Self.folderBookmarkKey)
+            }
+        }
+        return url
+    }
+
+    /// Store a folder URL as bookmark data. Returns the display path on success.
+    @discardableResult
+    public static func storeFolder(_ url: URL, defaults: UserDefaults = .standard) -> String? {
+        guard let data = try? url.bookmarkData() else { return nil }
```
[the remark this answers is not in the document]
Comment (moona3k) [reply Sources/MacParakeetCore/Services/AutoSaveService.swift]: Same as above — non-sandboxed app. `bookmarkData()` without security scope options is the correct call for Developer ID distribution.
